tests/Serializer/AbstractSerializerTest.php

```diff
@@ -65,6 +65,34 @@ public function testObjectsAreNotStrings(): void
         $this->assertSame(['key' => 'value'], $result);
     }
 
+    public function traversableDataProvider(): \Generator
+    {
+        yield [new \ArrayIterator(['foo', 'bar']), ['foo', 'bar']];
+
+        // Also test with a non-rewindable non-cloneable iterator:
+        $newGenerator = static function (array $array): \Generator {
+            yield from $array;
+        };
+
+        yield [$newGenerator(['foo', 'bar']), ['foo', 'bar']];
+    }
+
+    /**
+     * @dataProvider traversableDataProvider
+     */
+    public function testTraversablesAreNotConsumed(\Traversable $traversable, array $array): void
```

Comment:
As previously we were serializing any `iterable` types I think that it would be more approprite to change the typehint accordingly. Also :pray: change `$array` to `$input` to keep consistency with the rest of the test methods of this class